sdk/cosmos/azure_data_cosmos/src/routing/global_endpoint_manager.rs

```diff
@@ -0,0 +1,614 @@
+/// refreshing account properties, and resolving service endpoints based on request characteristics
+/// and availability. It handles endpoint discovery, tracks unavailable endpoints, and supports
+/// multi-master write configurations.
+#[derive(Debug, Clone)]
+pub struct GlobalEndpointManager {
+    /// The primary default endpoint URL for the Cosmos DB account
+    default_endpoint: String,
+
+    /// Thread-safe cache of location information including read/write endpoints and availability status
+    location_cache: Arc<Mutex<LocationCache>>,
+
+    /// HTTP pipeline for making requests to the Cosmos DB service
+    pipeline: Pipeline,
+
+    /// Cache for account properties with 600 second TTL to reduce redundant service calls
+    account_properties_cache: Cache<&'static str, AccountProperties>,
+}
+
+impl GlobalEndpointManager {
+    /// Creates a new `GlobalEndpointManager` with a `LocationCache` initialized
+    /// from the provided `default_endpoint` and `preferred_locations`.
+    ///
+    /// # Summary
+    /// Initializes the endpoint manager with a default endpoint, preferred regions for routing,
+    /// and an HTTP pipeline for communication. Sets up location cache for endpoint management
+    /// and account properties cache with 600 second TTL. The manager starts with empty endpoint
+    /// lists until the first account properties refresh populates regional endpoints.
+    ///
+    /// # Arguments
+    /// * `default_endpoint` - The primary Cosmos DB account endpoint URL
+    /// * `preferred_locations` - Ordered list of preferred Azure regions for request routing
+    /// * `pipeline` - HTTP pipeline for making service requests
+    ///
+    /// # Returns
+    /// A new `GlobalEndpointManager` instance ready for request routing
+    pub fn new(
+        default_endpoint: String,
+        preferred_locations: Vec<Cow<'static, str>>,
+        pipeline: Pipeline,
+    ) -> Self {
+        let location_cache = Arc::new(Mutex::new(LocationCache::new(
+            default_endpoint.clone(),
+            preferred_locations.clone(),
+        )));
+        let account_properties_cache = Cache::builder()
+            .max_capacity(1)
+            .time_to_live(std::time::Duration::from_secs(600))
+            .build();
+
+        Self {
+            default_endpoint,
+            location_cache,
+            pipeline,
+            account_properties_cache,
+        }
+    }
+
+    /// Returns the default hub endpoint URL for the Cosmos DB account.
+    ///
+    /// # Summary
+    /// Retrieves the primary endpoint URL that was configured during manager initialization.
+    /// This is the main entry point for the Cosmos DB account and is used as a fallback
+    /// when no preferred regional endpoints are available or configured.
+    ///
+    /// # Returns
+    /// The default endpoint URL as a String
+    pub fn get_hub_uri(&self) -> String {
+        self.default_endpoint.clone()
+    }
```

Comment:
Rust "getters" don't usually have a `get_` prefix. Also, this can just return a `&str` and avoid the clone:


```suggestion
    pub fn hub_uri(&self) -> &str {
        &self.default_endpoint
    }
```

Reply:
Refactored.